service/src/main/java/uk/gov/hmcts/reform/fpl/service/CaseInitiationService.java

```diff
@@ -0,0 +1,118 @@
+    private final FeatureToggleService featureToggleService;
+    private final LocalAuthorityService localAuthorityService;
+
+    public Optional<DynamicList> getOutsourcingLocalAuthoritiesDynamicList() {
+        final List<LocalAuthority> outsourcingLAs = getOutsourcingLocalAuthorities()
+            .stream()
+            .sorted(comparing(LocalAuthority::getName))
+            .collect(Collectors.toList());
+
+        if (isEmpty(outsourcingLAs)) {
+            return empty();
+        }
+        return ofNullable(dynamicLists.asDynamicList(outsourcingLAs, LocalAuthority::getCode, LocalAuthority::getName));
+    }
+
+    public List<String> checkUserAllowedToCreateCase(CaseData caseData) {
+        final List<String> errors = new ArrayList<>();
+
+        final String localAuthorityCode = getLocalAuthorityCode(caseData);
+
+        if (!featureToggleService.isCaseCreationForNotOnboardedUsersEnabled(localAuthorityCode)
+            && organisationService.findOrganisation().isEmpty()) {
+            errors.add("Register for an account.");
+            errors.add("You cannot start an online application until you’re fully registered.");
+            errors.add("Ask your local authority’s public law administrator, or email MyHMCTSsupport@justice.gov.uk, "
+                + "for help with registration.");
+        }
+
+        return errors;
+    }
+
+    public CaseData updateOrganisationsDetails(CaseData caseData) {
+        final String currentUserOrganisationId = organisationService.findOrganisation()
+            .map(Organisation::getOrganisationIdentifier)
+            .orElse(null);
+
+        final Optional<String> outsourcingLA = dynamicLists.getSelectedValue(caseData.getOutsourcingLAs());
+
+        if (outsourcingLA.isPresent()) {
+            String outsourcingOrgId = localAuthorityService.getLocalAuthorityId(outsourcingLA.get());
+
+            return caseData.toBuilder()
+                .outsourcingPolicy(organisationPolicy(currentUserOrganisationId, EPSMANAGING))
+                .localAuthorityPolicy(organisationPolicy(outsourcingOrgId, LASOLICITOR))
+                .caseLocalAuthority(outsourcingLA.get())
+                .caseLocalAuthorityName(localAuthorityService.getLocalAuthorityName(outsourcingLA.get()))
+                .build();
+        } else {
+            return caseData.toBuilder()
+                .localAuthorityPolicy(organisationPolicy(currentUserOrganisationId, LASOLICITOR))
+                .caseLocalAuthority(localAuthorityService.getLocalAuthorityCode())
+                .caseLocalAuthorityName(localAuthorityService
+                    .getLocalAuthorityName(localAuthorityService.getLocalAuthorityCode()))
+                .build();
+        }
+    }
+
+    public void grantCaseAccess(CaseData caseData) {
+        final Long caseId = caseData.getId();
+
+        if (nonNull(caseData.getOutsourcingPolicy())) {
+            caseAccessService.revokeCaseRoleFromUser(caseId, requestData.userId(), CREATOR);
```

Comment:
curiosity: did we find out why the CREATOR role is an issue and has to be removed?

Reply:
yeah, if there is single user who has access to a case and have CREATOR role then supplementary data is not populated and sharing a case is not possible

https://tools.hmcts.net/confluence/display/RCCD/A+Guide+to+Assign+Access+to+cases+for+professional+users%3A+configuration